### PR TITLE
refactor-2

### PR DESCRIPTION
**Pull request by sarab96**
class for xma session. refactor for cpp

### CODE REVIEW

File: src/xma/include/lib/xma_session.hpp [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    /** Private stats data attached to a specific session. This field is
+    allocated and managed by XMA for each session type. */
+    void          *stats;
+  
+
+  session(int s_id, XmaSessionType s_type, int32_t c_id);
+
+public:
+  int32_t
+  init(int32_t dev_idx, int32_t cu_idx, std::string cu_name);
```
Comment (stsoe) [inline src/xma/include/lib/xma_session.hpp:56]: const std::string&
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#Rf-in

File: src/xma/include/lib/xma_session.hpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+namespace xma_core {
+namespace plg {
+
+/**
+ * class session : abstraction for xma session
+ */
+class session
+{
+protected:
```
Comment (stsoe) [inline src/xma/include/lib/xma_session.hpp:34]: I am wary about protected.  But occasionally it is needed and make sense, but I don't have the context here.
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#Rnr-protected-data

File: src/xma/src/xmaplugin/xma_session.cpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+namespace xma_core {
+namespace plg {
+
+session::session(int s_id, XmaSessionType s_type, int32_t c_id)
+{
+  session_id = s_id;
+  session_type = s_type;
+  channel_id = c_id;
+  session_signature = nullptr;
```
Comment (stsoe) [inline src/xma/src/xmaplugin/xma_session.cpp:27]: Prefer in class initialization, in this case all ` = nullptr` can be in class.
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#Rc-in-class-initializer

File: src/xma/src/xmaplugin/xma_session.cpp [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+#include "lib/xma_session.hpp"
+#include "app/xmaerror.h"
+
+namespace xma_core {
+namespace plg {
+
+session::session(int s_id, XmaSessionType s_type, int32_t c_id)
```
Comment (stsoe) [inline src/xma/src/xmaplugin/xma_session.cpp:22]: Prefer initialization to assignment:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#Rc-initialize

**Review by stsoe — CHANGES_REQUESTED** [review]

File: src/xma/include/lib/xma_session.hpp (hunk 1 above)
Comment (sarab96) [reply src/xma/include/lib/xma_session.hpp]: Done

File: src/xma/src/xmaplugin/xma_session.cpp (hunk 3 above)
Comment (sarab96) [reply src/xma/src/xmaplugin/xma_session.cpp]: Agree. Done

File: src/xma/src/xmaplugin/xma_session.cpp (hunk 4 above)
Comment (sarab96) [reply src/xma/src/xmaplugin/xma_session.cpp:22]: Agree. Done

File: src/xma/include/lib/xma_session.hpp (hunk 2 above)
Comment (sarab96) [reply src/xma/include/lib/xma_session.hpp]: Thought about it. Use of private and getter/setter is better. Done

**Review by stsoe — APPROVED** [review]

**Pull request merged**